spec/integration/dry/configurable/setting_spec.rb

```diff
@@ -323,6 +331,22 @@
       expect(object.config.path).not_to be(new_object.config.path)
     end
 
+    it "makes only settings defined before instantiation available" do
+      klass.setting :before, default: "defined before"
+
+      object_1 = klass.new
+
+      klass.setting :after, default: "defined after"
+
+      object_2 = klass.new
+
+      expect(object_1.config.before).to eq "defined before"
+      expect(object_1.config).not_to respond_to(:after)
+
+      expect(object_2.config.before).to eq "defined before"
+      expect(object_2.config.after).to eq "defined after"
+    end
+
```

Comment:
n.b. **This is not new behaviour** for this PR. I've copied this example into the master branch, and it passes there also. However, I thought it was useful to specify this behaviour clearly given we have slightly different behaviour at the class-level now.